tests/patreon.js

```diff
@@ -13,26 +13,57 @@ test('patreon', (assert) => {
                 'Authorization header should be "Bearer token"'
             )
 
-            return { user: 'test' }
+            return {
+                data: {
+                    type: 'user',
+                    id: '123',
+                    attributes: {
+                        full_name: 'Test User'
+                    },
+                    relationships: {
+                        campaign: {
+                            data: {
+                                type: 'campaign',
+                                id: '456'
+                            }
+                        }
+                    }
+                },
+                included: [{
+                    type: 'campaign',
+                    id: '456',
+                    attributes: {
+                        'pledge_sum': 123456
+                    }
+                }]
+            }
         })
 
     const client = patreon('token')
 
     client('/current_user')
-        .then((res) => {
-            assert.ok(res, 'res should be a parsed json object')
-            assert.equal(res.user, 'test', 'res.test should equal "test"')
+        .then(({ store, rawJson }) => {
+            assert.equal(store.find('user', '123').full_name, 'Test User', 'store should be a JSON:API data store')
+            assert.equal(store.find('user', '123').campaign.pledge_sum, 123456, 'store should be a JSON:API data store')
+
+            assert.ok(rawJson, 'rawJson should be a parsed rawJson object')
+            assert.equal(rawJson.data.attributes.full_name, 'Test User', 'rawJson should have the correct content')
+
+            const _store = client.getStore()
+            console.log('_st', JSON.stringify(_store.findAll('user').map(user => user.serialize())))
+            assert.equal(_store.find('user', '123').full_name, 'Test User', 'store should be a JSON:API data store')
+            assert.equal(_store.find('user', '123').campaign.pledge_sum, 123456, 'store should be a JSON:API data store')
         })
         .catch((err) => {
-            assert.fail('promise failed unexpectedly!')
+            assert.fail(err, 'promise failed unexpectedly!')
         })
 
     nock('https://api.patreon.com')
         .get('/oauth2/api/current_user')
         .replyWithError('Oh geeze')
 
     client('/current_user')
-        .then((res) => {
+        .then((result) => {
```

Comment:
`res` intended abbreviation for `response` and now I will never use `res` again ;p

Reply:
yea i moved it to `result` because it's now more than just the `response` :+1: